dnf-behave-tests/features/dnf-transaction-exception.feature

```diff
@@ -0,0 +1,15 @@
+@no_installroot
+Feature: Test exception when dnf tries to do a transaction resolved but not valid anymore
+
+Background: setup repository
+  Given I use repository "dnf-transaction"
+    And I configure dnf with
+        | key          | value                                 |
+        | pluginpath   | /dnf-plugins |
+    And I do not disable plugins
+
+
+Scenario: Dnf fails to update a plugin if it does not exist after resolving the transaction
+  Given I successfully execute dnf with args "install wilee"
+   When I execute dnf with args "remove update wilee"
+   Then the exit code is 0
```

Comment:
You should probably do more on the result checking front, i.e. making sure what you expect to have transpired really has transpired?

Reply:
The test should check that "parsed dnf output transaction" contains `llama` package removal and the rpm transaction is empty (not sure we have a step for it)? Or what is the proposed fix in the dnf? I mean should it return 0 / nothing to do (because the package that was expected to be removed was already removed from outside), or should it return an error that system state during the transaction does not correspond to system state in transaction resolution time?

Reply:
Well, good question I guess :slightly_smiling_face: I am not sure at all, but I would lean towards the error state. I'm not sure about the real-world use case semantic details (what is the practical situation of rpm being called to remove the package at the same time dnf is running), but it is something unexpected and we are assuming a simple scenario. If there's actually something a bit more complex resulting in this, the user should be notified of the unusual situation, rather than risking it being ignored.

Reply:
That's also what I would prefer.
Just return nice error message instead of current ugly traceback.

Reply:
@lukash @m-blaha  thanks for the help, now the test is fine and is able to reproduce and test the error. the solution in my opinion is the same as Marek's. simply avoid the crash and return a dnf error. I am preparing the code now for the test and dnf as well